torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1196,7 +1196,7 @@ namespace {
 // Check validity of tensor
 void check_gpu_single_tensor(const at::Tensor& tensor) {
   if (!tensor.is_cuda() || tensor.is_sparse()) {
```

Comment:
checks like this should be pretty straightforward to just add cases for .is_lazy() || .is_xla()